### PR TITLE
prefabs, navmesh, spawning

### PR DESCRIPTION
**Pull request by ethanakester**
- Converted buildings + trees in the level into prefabs
- Created navmesh of whole level
- Enemy spawning is good (not clipping in buildings or trees or anything)

### CODE REVIEW

File: Assets/Scripts/EnemySpawn.cs [hunk 1]
```diff
@@ -38,14 +41,34 @@ void Update()
...
     void SpawnEnemy()
     {
-        //TODO: consider if we want to preset Enemy data in Prefab or upon method call
-        _enemySpawn.x = _platformTransform.position[0] + Random.Range(-_platformSize.x, _platformSize.x);
-        _enemySpawn.z = _platformTransform.position[2] + Random.Range(-_platformSize.z, _platformSize.z);
-        _enemySpawn.y = _platformTransform.position[1] + _platformSize.y;
+        Debug.Log("spawning");
+        // _enemySpawn.x = _platformTransform.position[0] + Random.Range(-_platformSize.x, _platformSize.x);
+        // _enemySpawn.z = _platformTransform.position[2] + Random.Range(-_platformSize.z, _platformSize.z);
+        // _enemySpawn.y = _platformTransform.position[1] + _platformSize.y;
```
Comment (MaXeraph) [inline Assets/Scripts/EnemySpawn.cs:65]: REMOVE

**Review by MaXeraph — APPROVED** [review]

File: Assets/Scripts/EnemySpawn.cs [hunk 2]
```diff
@@ -13,6 +14,8 @@ public class EnemySpawn : MonoBehaviour
     private const float _abovePlatformHeight = 0.6f; //TODO:  check this value
     private const float _platformOffset = 1f; //TODO:  check this value
 
+    private const float platformRadius = 175/2;
```
Comment (Sirkular) [inline Assets/Scripts/EnemySpawn.cs:17]: This should be mutable if we decide to create rooms of varying sizes.

File: Assets/Scripts/EnemySpawn.cs (hunk 1 above)
Comment (zhuandy42) [reply Assets/Scripts/EnemySpawn.cs:65]: REMOVE

File: Assets/Scripts/EnemySpawn.cs [hunk 3]
```diff
@@ -38,14 +41,34 @@ void Update()
...
     }
 
+    bool RandomPoint(float radius, out Vector3 result)
+    {
+        Vector3 randomPoint = Random.insideUnitSphere * radius;
+        NavMeshHit hit;
+        if (NavMesh.SamplePosition(randomPoint, out hit, radius, NavMesh.AllAreas))
+        {
+            result = hit.position;
+            Debug.Log(result);
```
Comment (Sirkular) [inline Assets/Scripts/EnemySpawn.cs:51]: Extra `Debug.log`s

File: Assets/Scripts/EnemySpawn.cs [hunk 4]
```diff
@@ -38,14 +41,34 @@ void Update()
...
+        Debug.Log("spawning");
+        // _enemySpawn.x = _platformTransform.position[0] + Random.Range(-_platformSize.x, _platformSize.x);
+        // _enemySpawn.z = _platformTransform.position[2] + Random.Range(-_platformSize.z, _platformSize.z);
+        // _enemySpawn.y = _platformTransform.position[1] + _platformSize.y;
 
-        GameObject.Instantiate(Resources.Load("Enemy"), _enemySpawn, Quaternion.identity);
+        Vector3 targetSpawn;
+        if (RandomPoint(platformRadius, out targetSpawn)) {
+            Debug.DrawRay(targetSpawn, Vector3.up, Color.blue, 1.0f);
+            GameObject.Instantiate(Resources.Load("Enemy"), targetSpawn, Quaternion.identity);
```
Comment (Sirkular) [inline Assets/Scripts/EnemySpawn.cs:70]: "Enemy" should be an argument, fixable later.

**Review by Sirkular — APPROVED** [review]

**Review by zhuandy42 — APPROVED** [review]

**Pull request merged**